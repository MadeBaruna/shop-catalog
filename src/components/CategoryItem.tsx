import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Div = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin: 0 23px 10px 23px;

  img {
    width: 64px;
    height: 64px;
    border-radius: 32px;
    margin-bottom: 4px;
  }

  p {
    text-transform: uppercase;
    font-size: 12px;
  }
`;

interface IProps {
  id: string;
  name: string;
  pictureUrl: string;
}

const CategoryItem: React.FC<IProps> = ({ id, name, pictureUrl }) => (
  <Link
    to={{
      pathname: `/products/${id}`,
      state: {
        title: name
      }
    }}
  >
    <Div>
      <img src={pictureUrl} alt="gambar kategori" />
      <p>{name}</p>
    </Div>
  </Link>
);

export default CategoryItem;
