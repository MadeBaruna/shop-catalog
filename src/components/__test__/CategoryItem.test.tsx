import React from 'react';

import CategoryItem from '../CategoryItem';
import { shallow } from 'enzyme';

it('renders <CategoryItem />', () => {
  const wrapper = shallow(
      <CategoryItem id="1" name="Midi Dress" pictureUrl="http://a.jpg" />
  );

  expect(wrapper).toMatchSnapshot();
});
