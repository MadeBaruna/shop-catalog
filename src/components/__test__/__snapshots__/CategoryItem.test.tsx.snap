// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <CategoryItem /> 1`] = `
<Link
  to={
    Object {
      "pathname": "/category/1",
      "state": Object {
        "title": "Midi Dress",
      },
    }
  }
>
  <styled.div>
    <img
      alt="gambar kategori"
      src="http://a.jpg"
    />
    <p>
      Midi Dress
    </p>
  </styled.div>
</Link>
`;
